internal/factsengine/adapters/rabbitmq.go

```diff
@@ -0,0 +1,79 @@
+package adapters
+
+import (
+	log "github.com/sirupsen/logrus"
+	"github.com/wagslane/go-rabbitmq"
+)
+
+const (
+	gatherFactsExchanage string = "gather_facts"
+	factsExchanage       string = "facts"
+)
+
+type rabbitMQAdapter struct {
+	consumer  rabbitmq.Consumer
+	publisher *rabbitmq.Publisher
+}
+
+func NewRabbitMQAdapter(factsEngineService string) *rabbitMQAdapter {
+	consumer, err := rabbitmq.NewConsumer(
+		factsEngineService,
+		rabbitmq.Config{},
+		rabbitmq.WithConsumerOptionsLogging,
+	)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	publisher, err := rabbitmq.NewPublisher(
+		factsEngineService,
+		rabbitmq.Config{},
+		rabbitmq.WithPublisherOptionsLogging,
+	)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	return &rabbitMQAdapter{
+		consumer:  consumer,
+		publisher: publisher,
+	}
+}
+
+func (r *rabbitMQAdapter) Unsubscribe() error {
+	r.consumer.Close()
```

Comment:
Could we check the errors returned by Close? We can return it and spot some problems

Reply:
Done!